indico/modules/events/registration/controllers/__init__.py

```diff
@@ -36,3 +41,22 @@ def _checkParams(self):
                         .find(id=request.view_args['reg_form_id'], is_deleted=False)
                         .options(defaultload('form_items').joinedload('children').joinedload('current_data'))
                         .one())
+
+
+class RegistrationCreationMixin(object):
+    """Mixin for a single registration RH"""
```

Comment:
Mixin for registration creation.

Actually, now that I think about it again, maybe a util function `create_registration(regform, data)` would be cleaner?